src/dispatch/plugins/dispatch_slack/workflow.py

```diff
@@ -0,0 +1,279 @@
+        artifact_links = ""
+        for a in w.artifacts:
+            artifact_links += f"- <{a.weblink}|{a.name}> \n"
+
+        blocks.append(
+            Section(
+                fields=[
+                    "*Name:* " + f"\n <{w.weblink}|{w.workflow.name}> \n"
+                    if w.weblink
+                    else "*Name:* " + f"\n {w.workflow.name} \n"
+                    f"*Workflow Description:* \n {w.workflow.description} \n"
+                    f"*Run Reason:* \n {w.run_reason} \n"
+                    f"*Creator:* \n {w.creator.individual.name} \n"
+                    f"*Status:* \n {w.status} \n"
+                    f"*Artifacts:* \n {artifact_links} \n"
+                ]
+            )
+        )
+
+    blocks = Message(blocks=blocks).build()["blocks"]
+    await respond(blocks=blocks, response_type="ephemeral")
+
+
+async def handle_workflow_run_command(
+    ack,
+    body,
+    client,
+    context,
+    db_session,
+):
+    """Handles the workflow run command."""
+    await ack()
+
+    blocks = [
+        Context(elements=[MarkdownText(text="Select a workflow to run.")]),
+        workflow_select(
+            db_session=db_session,
+            dispatch_action=True,
+        ),
+    ]
+
+    modal = Modal(
+        title="Run Workflow",
+        blocks=blocks,
+        submit="Run",
+        close="Close",
+        callback_id=RunWorkflowActions.submit,
+        private_metadata=context["subject"].json(),
+    ).build()
+
+    await client.views_open(view=modal, trigger_id=body["trigger_id"])
+
+
+@app.view(
+    RunWorkflowActions.submit,
+    middleware=[action_context_middleware, db_middleware, user_middleware, modal_submit_middleware],
+)
+async def handle_workflow_submission_event(ack, context, db_session, form_data, user):
+    """Handles workflow submission event."""
+    await ack()
+
+    incident = incident_service.get(db_session=db_session, incident_id=context["subject"].id)
+    workflow_id = form_data.get(RunWorkflowBlockIds.workflow_select)["value"]
+    workflow = workflow_service.get(db_session=db_session, workflow_id=workflow_id)
+
+    params = {}
+    named_params = []
+    for i in form_data.keys():
+        if i.startswith(RunWorkflowBlockIds.param_input):
+            key = i.split("-")[1]
+            value = form_data[i]
+            params.update({key: value})
+            named_params.append({"key": key, "value": value})
+
+    creator = participant_service.get_by_incident_id_and_email(
+        db_session=db_session, incident_id=incident.id, email=user.email
+    )
+
+    instance = workflow_service.create_instance(
+        db_session=db_session,
+        workflow=workflow,
+        instance_in=WorkflowInstanceCreate(
+            incident=incident,
+            creator=creator,
+            run_reason=form_data[RunWorkflowBlockIds.reason_input],
+            parameters=named_params,
+        ),
+    )
+
+    for p in instance.parameters:
+        if p["value"]:
```

Comment:
Should we guard against `KeyError` here?

Reply:
I don't think so, I would want the function to fail if the data structure was somehow missing or incomplete `value` is required for instance parameters.